core/src/main/scala/pureconfig/package.scala

```diff
@@ -97,11 +99,116 @@ package object pureconfig {
    *         `Config` from the configuration files, else a `Failure` with details on why it
    *         isn't possible
    */
-  def loadConfigWithFallback[Config](conf: TypesafeConfig, namespace: String)(implicit conv: ConfigConvert[Config]): Try[Config] = {
+  def loadConfigWithFallback[Config](conf: TypesafeConfig, namespace: String)(implicit conv: ConfigConvert[Config]): Either[ConfigReaderFailures, Config] = {
     ConfigFactory.invalidateCaches()
     loadConfig[Config](conf.withFallback(ConfigFactory.load()), namespace)
   }
 
+  private def getResultOrThrow[Config](failuresOrResult: Either[ConfigReaderFailures, Config])(implicit ct: ClassTag[Config]): Config = {
+    failuresOrResult match {
+      case Right(config) => config
+      case Left(failures) => throw new ConfigReaderException[Config](failures)
+    }
+  }
+
+  /**
+   * Load a configuration of type `Config` from the standard configuration files
+   *
+   * @return the configuration
+   */
+  @throws[ConfigReaderException[_]]
+  def loadConfigOrThrow[Config](implicit conv: ConfigConvert[Config], ct: ClassTag[Config]): Config = {
```

Comment:
It doesn't seem reasonable to ask for a `ClassTag` just for the purpose of printing a pretty error message. Have you considered using the `null` default implicit value trick, like:

```scala
def loadConfigOrThrow[Config](implicit conv: ConfigConvert[Config], ct: ClassTag[Config] = null): Config
```

If you define the method this way the compiler will provide a `ClassTag` when it can and you can still handle `null` class tags in a special way in `ConfigReaderException` - for example, by printing a generic header message instead of specifying the class name.